File: Irrgarten/lib/FuzzyPlayer.rb
require_relative 'Player'

class FuzzyPlayer < Player 
	
	def initialize(other) 
		copy(other)
	end
	
	def move(direction, validMoves)
		move = Dice.nextStep(super, validMoves, getIntelligence)
		return move
	end
	
	def attack
		return Dice.intensity(getStrength) + sumWeapons
	end
	
	def defensiveEnergy
		return Dice.intensity(getIntelligence) + sumShields
	end
	
	def to_s
		return "\nFuzzy" + super
	end
	
	protected :defensiveEnergy
end

File: irrgarten/src/irrgarten/FuzzyPlayer.java
package irrgarten;

import java.util.ArrayList;

/**
 *
 * @author usuario
 */

//HAY QUE VER DONDE SE HACE REFERENCIA A PLAYER E IMPLEMENTAR PARA QUE PUEDA SER 
//SUSTITUIDO POR FUZZYPLAYER 
//(Clase game, array list de jugadores)
public class FuzzyPlayer extends Player{
    FuzzyPlayer(Player other){
        super(other);
    }
    
    @Override
    public Directions move(Directions direction, ArrayList<Directions>  validMoves ){
        Directions dir = super.move(direction, validMoves);
        Directions move = Dice.nextStep(dir, validMoves, this.getIntelligence());
        return move;
    }
    
    @Override
    public float attack(){       
        return Dice.intensity(this.getStrength()) + sumWeapons();
    } 
    
    @Override
    protected float defensiveEnergy(){
        return Dice.intensity(this.getIntelligence()) + sumShields();
    }
    
    @Override
    public String toString(){
        return "\nFuzzy" + super.toString();
    }
    
    
}
